### PR TITLE
fixes in smc backend refactoring, added test

### PR DESCRIPTION
**Pull request by hvasbath**
Fixes to refactoring from @ColCarrol , added test that tests stage_handler features.

### CODE REVIEW

File: pymc3/tests/test_smc.py [hunk 1]
```diff
@@ -13,15 +14,13 @@
 @pytest.mark.xfail(condition=(theano.config.floatX == "float32"), reason="Fails on float32")
 class TestSMC(SeededTest):
 
-    def setup_method(self):
-        super(TestSMC, self).setup_method()
+    def setup_class(self):
```
Comment (ColCarroll) [inline pymc3/tests/test_smc.py:17]: just want to make sure this is intentional -- you'll now reuse the same test folder for every test in the class

File: pymc3/backends/smc_text.py [hunk 2]
```diff
@@ -196,8 +196,14 @@ def load_atmip_params(self, stage_number):
         else:
             prev = stage_number - 1
         pm._log.info('Loading parameters from completed stage {}'.format(prev))
-        with open(self.atmip_path(prev), 'rb') as buff:
-            return pickle.load(buff)
+
+        with model:
```
Comment (ColCarroll) [inline pymc3/backends/smc_text.py:200]: I like that this nesting is explicit, but just for interest's sake:

starting with python 2.7 you can use multiple context managers in one line
```
with open(...) as buff, model:
    step = pickle.load(buff)
```

**Review by ColCarroll — COMMENTED** [review]
LGTM, will merge when tests pass

File: pymc3/tests/test_smc.py (hunk 1 above)
Comment (hvasbath) [reply pymc3/tests/test_smc.py:17]: ah good point! will fix that!
edit: no fix needed, as we set the rm_flag=True it removes the previous results

File: pymc3/backends/smc_text.py (hunk 2 above)
Comment (hvasbath) [reply pymc3/backends/smc_text.py:200]: cool! didnt know that! but here I think it is indeed better to have them nested for clarity

**Comment by ColCarroll** [comment]
thanks!

**Pull request merged**